assets/js/components/DashboardMainApp.js

```diff
@@ -157,12 +162,12 @@ export default function DashboardMainApp() {
 				accidentally rendering the widget area if any child widgets accidentally
 				render when `userInputEnabled` is false.
 
-				This check can be removed once the User Input feature is fully launched
+				The userInputEnabled check can be removed once the User Input feature is fully launched
 				and we remove this feature flag.
 
 				See: https://github.com/google/site-kit-wp/pull/6630#discussion_r1127229162
 			*/ }
-			{ userInputEnabled && (
+			{ userInputEnabled && isKeyMetricsWidgetHidden !== true && (
```

Comment:
Hey @jimmymadon, nice work on this PR.

Looking at this condition, I'm wondering if it should be tweaked so the `WidgetContextRenderer` is not rendered when `isKeyMetricsWidgetHidden` is `undefined`, i.e. the state is still resolving. To be honest in my testing this seems to be working fine in practice, so I don't want to throw a red herring in here, but it feels like the safer thing to do would be avoid rendering these widgets at the top level until the `isKeyMetricsWidgetHidden` state is resolved, so we don't initially render them and then remove once the `isKeyMetricsWidgetHidden` is resolved to `true`.

In other words we could update as follows:

```suggestion
			{ userInputEnabled && isKeyMetricsWidgetHidden === false && (
```

Have you already considered this, and what do you think?

Reply:
Hey @jimmymadon, thanks for the background here and for making the change. However, and I must apologise because I could have saved us some time here if I'd only spotted this the first time around - this change does actually introduce a bug, whereby the Key Metrics navigation chip is no longer correctly selected on page load. Here's a screen capture to demonstrate:

https://github.com/google/site-kit-wp/assets/18395600/5bb271b5-06bb-404b-8dcd-4dcb551bbee8

And, confirming the behaviour is correct upon reversion of the change:

https://github.com/google/site-kit-wp/assets/18395600/2bf9b31d-95f0-4ff5-8638-5a7838fef701

As a result, I'll go ahead and revert that commit, restoring the PR to how it was when I initially picked it up for MR. Considering that we've not seen any evidence of a flicker in our testing this seems like the way to go here, and we can just keep an eye out for it in future.